### ISSUE TITLE
Can't run on iOS after upgrading to Channel beta, 2.1.0-12.2.pre from Beta 2.0.1

### ISSUE THREAD

**Issue by imSaharukh** [flutter/flutter#80107]
I'm unable to run my app on iOS after upgrading to  2.1.0-12.2.pre from Beta 2.0.1
everything was perfectly fine before upgrading

**Expected results:** 
it should run perfectly 

**Actual results:** 
Failed to build iOS app
<details>
  <summary>Logs</summary>

```
Error output from Xcode build:
↳
    ** BUILD FAILED **
Xcode's output:
↳
    Command PhaseScriptExecution failed with a nonzero exit code
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description

               [ +113 ms] Initializing file store
               [  +17 ms] Done initializing file store
               [  +94 ms] debug_unpack_ios: Starting due to {}
               [  +20 ms] Skipping target: gen_localizations
               [ +684 ms] Skipping target: kernel_snapshot
               [  +10 ms] Persisting file store
               [  +13 ms] Done persisting file store
               [  +13 ms] Target debug_unpack_ios failed: Exception: Binary /Volumes/Work/Cookspace/Flutter App/flutter-app/build/ios/Debug-iphonesimulator/Flutter.framework/Flutter does not contain arm64 x86_64. Running lipo -info:
                          Non-fat file: /Volumes/Work/Cookspace/Flutter App/flutter-app/build/ios/Debug-iphonesimulator/Flutter.framework/Flutter is architecture: x86_64

                          #0      UnpackIOS._thinFramework (package:flutter_tools/src/build_system/targets/ios.dart:343:7)
                          #1      UnpackIOS.build (package:flutter_tools/src/build_system/targets/ios.dart:293:11)
                          <asynchronous suspension>
                          #2      _BuildInstance._invokeInternal (package:flutter_tools/src/build_system/build_system.dart:831:9)
                          <asynchronous suspension>
                          #3      Future.wait.<anonymous closure> (dart:async/future.dart)
                          <asynchronous suspension>
                          #4      _BuildInstance.invokeTarget (package:flutter_tools/src/build_system/build_system.dart:769:32)
                          <asynchronous suspension>
                          #5      Future.wait.<anonymous closure> (dart:async/future.dart)
                          <asynchronous suspension>
                          #6      _BuildInstance.invokeTarget (package:flutter_tools/src/build_system/build_system.dart:769:32)
                          <asynchronous suspension>
                          #7      FlutterBuildSystem.build (package:flutter_tools/src/build_system/build_system.dart:598:16)
                          <asynchronous suspension>
                          #8      AssembleCommand.runCommand (package:flutter_tools/src/commands/assemble.dart:290:32)
                          <asynchronous suspension>
                          #9      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1037:27)
                          <asynchronous suspension>
                          #10     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:152:19)
                          <asynchronous suspension>
                          #11     CommandRunner.runCommand (package:args/command_runner.dart:197:13)
                          <asynchronous suspension>
                          #12     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:283:9)
                          <asynchronous suspension>
                          #13     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:152:19)
                          <asynchronous suspension>
                          #14     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:239:5)
                          <asynchronous suspension>
                          #15     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:62:9)
                          <asynchronous suspension>
                          #16     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:152:19)
                          <asynchronous suspension>
                          #17     main (package:flutter_tools/executable.dart:92:3)
                          <asynchronous suspension>

               [  +20 ms] "flutter assemble" took 1,066ms.
               [   +4 ms]
                          #0      throwToolExit (package:flutter_tools/src/base/common.dart:12:3)
                          #1      AssembleCommand.runCommand (package:flutter_tools/src/commands/assemble.dart:307:7)
                          <asynchronous suspension>
                          #2      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1037:27)
                          <asynchronous suspension>
                          #3      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:152:19)
                          <asynchronous suspension>
                          #4      CommandRunner.runCommand (package:args/command_runner.dart:197:13)
                          <asynchronous suspension>
                          #5      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:283:9)
                          <asynchronous suspension>
                          #6      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:152:19)
                          <asynchronous suspension>
                          #7      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:239:5)
                          <asynchronous suspension>
                          #8      run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:62:9)
                          <asynchronous suspension>
                          #9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:152:19)
                          <asynchronous suspension>
                          #10     main (package:flutter_tools/executable.dart:92:3)
                          <asynchronous suspension>


               [   +2 ms] ensureAnalyticsSent: 0ms
               [   +1 ms] Running shutdown hooks
               [        ] Shutdown hooks complete
               [   +3 ms] exiting with code 1
               Command PhaseScriptExecution failed with a nonzero exit code


[  +23 ms] Could not build the application for the simulator.
[   +2 ms] Error launching application on iPhone 12 Pro Max.
[   +4 ms] "flutter run" took 17,749ms.
[   +4 ms]
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:12:3)
           #1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:659:9)
           <asynchronous suspension>
           #2      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1037:27)
           <asynchronous suspension>
           #3      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:152:19)
           <asynchronous suspension>
           #4      CommandRunner.runCommand (package:args/command_runner.dart:197:13)
           <asynchronous suspension>
           #5      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:283:9)
           <asynchronous suspension>
           #6      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:152:19)
           <asynchronous suspension>
           #7      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:239:5)
           <asynchronous suspension>
           #8      run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:62:9)
           <asynchronous suspension>
           #9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:152:19)
           <asynchronous suspension>
           #10     main (package:flutter_tools/executable.dart:92:3)
           <asynchronous suspension>


[ +100 ms] ensureAnalyticsSent: 99ms
[   +1 ms] Running shutdown hooks
[        ] Shutdown hook priority 4
[   +3 ms] Shutdown hooks complete
[        ] exiting with code 1
```

flutter doctor -v

```
[✓] Flutter (Channel beta, 2.1.0-12.2.pre, on macOS 11.2.3 20D91 darwin-x64,
    locale en-BD)
    • Flutter version 2.1.0-12.2.pre at /Volumes/Work/FlutterSrc/flutter
    • Framework revision 5bedb7b1d5 (3 weeks ago), 2021-03-17 17:06:30 -0700
    • Engine revision 711ab3fda0
    • Dart version 2.13.0 (build 2.13.0-116.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/imsaharukh/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.55.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (3 available)
    • iPhone 12 Pro Max (mobile) • 52369689-3A84-4B3B-84FF-F58F0F53F65B • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    • macOS (desktop)            • macos                                •
      darwin-x64     • macOS 11.2.3 20D91 darwin-x64
    • Chrome (web)               • chrome                               •
      web-javascript • Google Chrome 89.0.4389.114

• No issues found!
```

</details>

**Comment by TahaTesser** [comment]
Hi @imSaharukh 
Did you update your packages? Did you clean the project
Can you please provide complete `flutter run --verbose` in a text file
Thank you

**Comment by imSaharukh** [comment]
@TahaTesser Yes, I have updated all my packages and also clean the project 

here is my ```flutter run --verbose``` output

[output.txt](https://github.com/flutter/flutter/files/6287148/output.txt)

**Comment by imSaharukh** [comment]
i am able to run on ios by deleting ios folder and run ```flutter create .```
but didn't understand what's the issue

**Comment by TahaTesser** [comment]
Hi @imSaharukh 
It seems to be something with the build script, which fixed by generating a fresh folder. If the problem persists, please comment below.
Given your last message I feel safe to close this issue, if you disagree please write in the comments and I will reopen it. 
Thank you

**Comment by anhlevfx** [comment]
Hi @TahaTesser I'm facing the same problem. After upgrade to flutter 2.2.0, I can not run on iPhone simulator 14.5. But, when use XCode to run in debug mode, everything works fine.

**Comment by TahaTesser** [comment]
@anhlevfx 
please file a new issue with the exact description of what happens, logs, and `flutter doctor -v` output and most importantly reproducible steps

**Comment by imSaharukh** [comment]
> Hi @TahaTesser I'm facing the same problem. After upgrade to flutter 2.2.0, I can not run on iPhone simulator 14.5. But, when use XCode to run in debug mode, everything works fine.

try deleting the iOS folder (save the info.plist)
and recreate the ios folder using "flutter create ."

this trick worked for me on that time
